After enabling C++11, Ice fails to build.

Presumably, something in the old environment allowed
these source files to build. Perhaps another header
ensured that unistd.h was included here.

However, in the new environment, we have to ensure
unistd.h is included ourselves.

--- ./cpp/src/IceUtil/FileUtil.cpp
+++ ./cpp/src/IceUtil/FileUtil.cpp
@@ -26,7 +26,7 @@
 #  include <dir.h>
 #endif
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__) || defined(__linux__)
 #  include <unistd.h>
 #endif
 
--- ./cpp/src/Slice/FileTracker.cpp
+++ src/Slice/FileTracker.cpp
@@ -17,7 +17,7 @@
 #   include <direct.h>
 #endif
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__) || defined(__linux__)
 #   include <unistd.h>
 #endif
 
--- ./cpp/test/IceUtil/unicode/Client.cpp
+++ ./cpp/test/IceUtil/unicode/Client.cpp
@@ -15,7 +15,7 @@
 #   include <io.h>
 #endif
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__) || defined(__linux__)
 #   include <unistd.h>
 #endif
 
